test/powershell/Modules/Microsoft.PowerShell.Management/Wait-Process.Tests.ps1

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe "Wait-Process" {
+
+    BeforeAll {
+        $isNanoServer = [System.Management.Automation.Platform]::IsNanoServer
+        $isIot = [System.Management.Automation.Platform]::IsIoT
+        $isFullWin = $IsWindows -and !$isNanoServer -and !$isIot
+
+        $pingCommandPath = (Get-Command -CommandType Application ping)[0].Definition
+
+        $startProcessArgs = @{
+            FilePath = $pingCommandPath
+            PassThru = $true
+        }
+        if ($isFullWin) {
+            $startProcessArgs.WindowStyle = [System.Diagnostics.ProcessWindowStyle]::Hidden
+        }
+        $nc = $IsWindows ? "-n" : "-c"
+
+        function longPing { Start-Process @startProcessArgs -ArgumentList "$nc 10 localhost" }
+        function shortPing { Start-Process @startProcessArgs -ArgumentList "$nc 2 localhost" }
```

Comment:
Would it make sense to have a test hook in `Start-Process` to wait on actually starting the process until `Wait-Process` tells it?  This way we can cover test cases for processes that have already exited before `Wait-Process` starts and also after?

Reply:
@SteveL-MSFT, to make sure I understand correctly, are we talking about adding some tests where one or more of the processes has exited before `Wait-Process` is run, or did you have something more elaborate in mind?

Reply:
@dwtaber correct, having tests that cover the case where a process exited before `Wait-Process` has started and also test case where a process exists while `Wait-Process` is running.  You can add a test-hook to make this behavior predictable I think.

Reply:
We have Wait-Until helper function.

Reply:
It turns out that `Wait-Process -Any` was ignoring processes that ended before the wait started.  I believe I've fixed that issue, and I've added a few tests using `Wait-UntilTrue` (thanks for that suggestion!)